stan/math/rev/core/arena_matrix.hpp

```diff
@@ -110,23 +154,32 @@ class arena_matrix : public Eigen::Map<MatrixType> {
    * @param a expression to evaluate into this
    * @return `*this`
    */
-  template <typename T>
-  arena_matrix& operator=(const T& a) {
-    // do we need to transpose?
-    if ((RowsAtCompileTime == 1 && T::ColsAtCompileTime == 1)
-        || (ColsAtCompileTime == 1 && T::RowsAtCompileTime == 1)) {
-      // placement new changes what data map points to - there is no allocation
-      new (this) Base(
-          ChainableStack::instance_->memalloc_.alloc_array<Scalar>(a.size()),
-          a.cols(), a.rows());
-
+  template <typename T, require_not_arena_matrix_t<T>* = nullptr>
+  arena_matrix& operator=(T&& a) {
+    using T_t = std::decay_t<T>;
```

Comment:
Should this be `plain_type_t` instead of just decaying? I don't think all expressions will have a `ColsAtCompileTime`/`RowsAtCompileTime` that can be called